src/SFA.DAS.ApprenticeFeedback.Database/StoredProcedures/GenerateFeedbackTransactions.sql

```diff
@@ -1,23 +1,148 @@
-﻿CREATE PROCEDURE dbo.GenerateFeedbackTransactions 
+﻿CREATE PROCEDURE [dbo].[GenerateFeedbackTransactions]
 
-	@SentDateAgeDays int
+    @SentDateAgeDays int = 90
 
 AS
+BEGIN
+SET NOCOUNT ON;
 
-DECLARE @CreatedOn datetime
-SELECT @CreatedOn = GETUTCDATE()
-
-INSERT INTO dbo.FeedbackTransaction (ApprenticeFeedbackTargetId, CreatedOn)
-SELECT aft.Id, @CreatedOn
-FROM [dbo].[ApprenticeFeedbackTarget] aft
-LEFT JOIN (SELECT ApprenticeFeedbackTargetId
-           FROM [dbo].[FeedbackTransaction] 
-           WHERE SentDate IS NULL OR (SentDate IS NOT NULL AND SentDate >= DATEADD(day, 0 - @SentDateAgeDays, GETDATE())) ) ft1
-    on ft1.ApprenticeFeedbackTargetId = aft.Id
-WHERE Status = 2 -- "active"
-AND FeedbackEligibility = 1 -- "allow"
-AND ft1.ApprenticeFeedbackTargetId IS NULL
-
-SELECT @@ROWCOUNT AS Count, @CreatedOn AS CreatedOn
+DECLARE @Error_Code INT = 0
+        ,@Error_Message VARCHAR(MAX)
+        ,@Error_Severity INT = 0
+        
+    BEGIN TRANSACTION T1;
+    
+    
+    DECLARE @CreatedOn datetime, @count int;
+    SELECT @CreatedOn = GETUTCDATE();
```

Comment:
If the @CreatedOn was a parameter with default GETUTCDATE() or a default of null and set to GETUTCDATE when null then this can be used for testing the proc on various dates

Reply:
True. This was original code so was just going with it. But easy to change.